File: cinema-booking-client/src/interfaces/Booking.ts
import type { Seat } from "./Cinema";
import type { Showtime } from "./Showtime";

export const BookingStatus = {
  PENDING: 'PENDING',
  CONFIRMED: 'CONFIRMED',
  CANCELLED: 'CANCELLED',
  COMPLETED: 'COMPLETED',
} as const;

export type BookingStatus = typeof BookingStatus[keyof typeof BookingStatus];

export interface Booking {
  id: string;
  userId: string;
  showtimeId: string;
  seats: Seat[];
  totalPrice: number;
  status: BookingStatus;
  qrCode: string;
  createdAt: string;
  expiresAt: string;
  showtime?: Showtime;
}

File: cinema-booking-server/cinema-booking-server/Models/Booking.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace cinema_booking_server.Models
{
    public class Booking
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [ForeignKey("User")]
        public Guid? UserId { get; set; }

        [ForeignKey("Showtime")]
        [Required]
        public Guid ShowtimeId { get; set; }

        [ForeignKey("Payment")]
        public Guid? PaymentId { get; set; }

        [Required]
        [Column(TypeName = "decimal(12,2)")]
        public decimal TotalAmount { get; set; }

        [Required]
        [StringLength(50)]
        public string Status { get; set; } = "pending"; // pending, confirmed, cancelled

        [Required]
        public DateTimeOffset CreatedAt { get; set; } = DateTimeOffset.UtcNow;

        public DateTimeOffset? UpdatedAt { get; set; }

        // Navigation properties
        public User? User { get; set; }
        public Showtime? Showtime { get; set; }
        public Payment? Payment { get; set; }
        public ICollection<BookingSeat> BookingSeats { get; set; } = new List<BookingSeat>();
    }
}